File: Database/Model/Player.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Database.Model
{
    public class Player
    {
        [Key]
        [ForeignKey("User")]
        public int PlayerId { get; set; }
        public User User { get; set; }
        public string Province {  get; set; }
        public string City {  get; set; }
        public int TotalMatch { get; set; } = 0;
        public int TotalWins { get; set; } = 0;
        public int RankingPoint { get; set; } = 0;
        public int ExperienceLevel { get; set; } = 1;
        public DateTime JoinedAt { get; set; } = DateTime.UtcNow;

        public ICollection<TournamentRegistration> TournamentRegistrations { get; set; }
        public ICollection<Ranking> Rankings { get; set; } = new List<Ranking>();
        public ICollection<MatchesSendRequest> SentRequests { get; set; } = new List<MatchesSendRequest>();
        public ICollection<MatchesSendRequest> ReceivedRequests { get; set; } = new List<MatchesSendRequest>();
    }
}

File: MB/lib/core/models/player.dart
class Player {
  String province;
  String city;
  int totalMatch;
  int totalWins;
  int rankingPoint;
  int experienceLevel;
  DateTime joinedAt;

  Player({
    required this.province,
    required this.city,
    required this.totalMatch,
    required this.totalWins,
    required this.rankingPoint,
    required this.experienceLevel,
    required this.joinedAt,
  });

  factory Player.fromJson(Map<String, dynamic> json) {
    return Player(
      province: json['province'] as String,
      city: json['city'] as String,
      totalMatch: json['totalMatch'] as int,
      totalWins: json['totalWins'] as int,
      rankingPoint: json['rankingPoint'] as int,
      experienceLevel: json['experienceLevel'] as int,
      joinedAt: DateTime.parse(json['joinedAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'province': province,
      'city': city,
      'totalMatch': totalMatch,
      'totalWins': totalWins,
      'rankingPoint': rankingPoint,
      'experienceLevel': experienceLevel,
      'joinedAt': joinedAt.toIso8601String(),
    };
  }
}
